onset_maproom/calc.py

```diff
@@ -38,17 +38,25 @@ def estimate_sm(
         name="soil moisture",
         attrs=dict(description="Soil Moisture", units="mm"),
     )
-
+    # Case et doesn't depend on time_coord
+    if np.logical_not(type(et) is xr.core.dataarray.DataArray):
```

Comment:
I did this because isel({time_coord: i}) applied to et without a time_coord returned an error. But maybe there is a more elegant way to deal with that?

Reply:
Since et and daily_rain have the same dimension name and coordinate values for time, I think you can just do
```
        sminit + daily_rain.isel({time_coord: 0}) - et
```
and it will use the appropriate values from et and ignore the rest.

Reply:
didn't think of that... that makes all sense.

But it doesn't seem to behave that way... See last commit where et_has_T test fails and I included prints in the main function that I think describe the behavior.

Maybe there is an isel option to keep the dim?

Reply:
The [doc](https://xarray.pydata.org/en/stable/user-guide/computation.html#automatic-alignment) on alignment supports your diagnosis.
> Xarray enforces alignment between index Coordinates (that is, coordinates with the same name as a dimension, marked by *) on objects used in binary operations.

`T` is not an index coordinate in `daily_rain.isel(T=0)`, so it doesn't get aligned.

Your solution (subtract `daily_rain - et` before indexing) is succinct but probably inefficient. It repeats the same subtraction operation between the same two X * Y * T arrays on every iteration of the loop. To avoid that, we could pull out the subtraction operation and do it just once, before entering the loop. Alternatively, it looks like [expand_dims](https://xarray.pydata.org/en/stable/generated/xarray.DataArray.expand_dims.html) could convert the scalar `T` back to an index coordinate.

Sorry, I know it's annoying to wait five days to get feedback. If there are more things to discuss we should schedule time to work through them together.

Reply:
no problem. I can't really focus much on this these days.
I pushed the delta solution that seemed the most appropriate to me.